python/spear/__init__.py

```diff
@@ -4,7 +4,124 @@
+
+# This function returns a config object, obtained by loading and merging a list of config
+# files in the order they appear in the user_config_files input argument. This function is
+# useful for loading default values from multiple different components and systems, and then
+# overriding some of the default values with experiment-specific or user-specific overrides.
+# Before loading any of the config files specified in user_config_files, all the default
+# values required by the spear Python package and C++ plugins are loaded, and can be
+# overridden by any of the files appearing in user_config_files.
+def get_config(user_config_files):
+
+    # create a single CfgNode that will eventually contain data from all config files
+    config = CfgNode(new_allowed=True)
+
+    for c in default_config_files:
+        config.merge_from_file(c)
+
+    # In some cases, we need to update specific config values with information that is available
+    # at runtime, but isn't available when we are authoring our default_config.*.yaml files.
+    config.URDF_ROBOT.URDF_ROBOT_PAWN.URDF_DIR = os.path.realpath(os.path.join(spear_root_dir, "urdf"))
+
+    for c in user_config_files:
+        config.set_new_allowed(True)  # required to override an empty dict with a non-empty dict
+        config.merge_from_file(c)
+
+    config.freeze()
+
+    return config
+
+def configure_system(config):
+
+    # create a symlink to SPEAR.INSTANCE.PAKS_DIR
+    if config.SPEAR.INSTANCE.LAUNCH_MODE == "standalone" and config.SPEAR.INSTANCE.PAKS_DIR != "":
+
+        assert os.path.exists(config.SPEAR.INSTANCE.STANDALONE)
+        assert os.path.exists(config.SPEAR.INSTANCE.PAKS_DIR)
+
+        if sys.platform == "win32":
+            paks_dir = \
+                os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(config.SPEAR.INSTANCE.STANDALONE)), "..", "..", "Content", "Paks"))
+        elif sys.platform == "darwin":
+            paks_dir = \
+                os.path.realpath(os.path.join(config.SPEAR.INSTANCE.STANDALONE, "Contents", "UE", "SpearSim", "Content", "Paks"))
+        elif sys.platform == "linux":
+            paks_dir = \
+                os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(config.SPEAR.INSTANCE.STANDALONE)), "SpearSim", "Content", "Paks"))
+        else:
+            assert False
+
+        assert os.path.exists(paks_dir)
+
+        # we don't use os.path.realpath here because we don't want to resolve the symlink
+        spear_paks_dir = os.path.join(paks_dir, "SpearPaks")
+
+        if path_exists(spear_paks_dir):
+            log(f"File or directory or symlink exists, removing: {spear_paks_dir}")
+            remove_path(spear_paks_dir)
+
+        log(f"Creating symlink: {spear_paks_dir} -> {config.SPEAR.INSTANCE.PAKS_DIR}")
+        os.symlink(config.SPEAR.INSTANCE.PAKS_DIR, spear_paks_dir)
+
+    # provide additional control over which Vulkan devices are recognized by Unreal
```

Comment:
Let's do the same thing for environment variables as we did for command-line arguments. Let them be their own dictionary in the config. That way, users can set any number of environment variables to be whatever they want, and we don't need overly specific logic like this in our code.